### PR TITLE
feat(example): add Curve to `explorer` example

### PR DESCRIPTION
**Pull request by zizou0x**
The purpose of this PR is to add Curve support in the `explorer` example. 

It also contains a few small fixes that are needed to support Curve.

### CODE REVIEW

File: src/evm/protocol/vm/state.rs [hunk 1]
```diff
@@ -219,13 +219,13 @@ where
         tokens: Vec<Address>,
         max_amount: U256,
     ) -> Result<HashMap<Address, Overwrites>, SimulationError> {
-        let sell_token = &tokens[0].clone();
+        let sell_token = &tokens[0].clone(); //TODO: need to make it clearer from the interface
```
Comment (zizou0x) [inline src/evm/protocol/vm/state.rs:222]: I don't think this PR is the correct place to change this so I added a TODO.

File: src/evm/protocol/vm/state_builder.rs [hunk 2]
```diff
@@ -253,8 +253,8 @@ where
 
         if let Some(stateless_contracts) = &self.stateless_contracts {
             for (address, bytecode) in stateless_contracts.iter() {
+                let mut addr_str = address.clone();
```
Comment (dianacarvalho1) [inline src/evm/protocol/vm/state_builder.rs:256]: ah shit, good catch! 🙏🏼

File: src/evm/protocol/vm/utils.rs [hunk 3]
```diff
@@ -350,22 +355,161 @@ pub fn hexstring_to_vec(hexstring: &str) -> Result<Vec<u8>, SimulationError> {
...
+/// ```
+/// use json_deserialize_address_list;
+///
+/// let json_input = br#"["0x1234", "0xc0ffee"]"#;
+/// match json_deserialize_address_list(json_input) {
+///     Ok(result) => println!("Decoded: {:?}", result),
+///     Err(e) => eprintln!("Error: {}", e),
+/// }
+/// ```
+pub fn json_deserialize_address_list(input: &[u8]) -> Result<Vec<Vec<u8>>, SimulationError> {
```
Comment (dianacarvalho1) [inline src/evm/protocol/vm/utils.rs:434]: where is this being used??

**Review by dianacarvalho1 — APPROVED** [review]
Thank you so much @zizou0x 🙏🏼 only have 1 small question

File: src/evm/protocol/vm/utils.rs (hunk 3 above)
Comment (zizou0x) [reply src/evm/protocol/vm/utils.rs:434]: It's the deserialize function related to [this serialize function](https://github.com/search?q=repo:propeller-heads/tycho-protocol-sdk%20json_serialize_address_list&type=code) that we have in the SDK. 

Currently it's only used by Balancer for rate providers.

**Pull request merged**